script.js

```diff
@@ -1,4 +1,214 @@
+
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter; // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+
+      if (cardName == 1) {
+        cardName = "Ace";
+      } else if (cardName == 11) {
+        cardName = "Jack";
+      } else if (cardName == 12) {
+        cardName = "Queen";
+      } else if (cardName == 13) {
+        cardName = "King";
+      }
+      // Create a new card with the current name, suit, and rank
+      //P1: Cards are objects containing (name,suit,rank)
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      }; // Add the new card to the deck
+
+      cardDeck.push(card); // Increment rankCounter to iterate over the next rank
+
+      rankCounter += 1;
+    } // Increment the suit index to iterate over the next suit
+
+    suitIndex += 1;
+  } // Return the completed card deck
+
+  return cardDeck;
+};
+
+//P1: Create a Function for shuffling the deck.
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length); // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex]; // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex]; // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard; // Increment currentIndex to shuffle the next pair of cards
+    currentIndex += 1;
+  } // Return the shuffled deck
+  return cardDeck;
+};
+var deck = shuffleCards(makeDeck());
+
+//Global Variables
+//Default Mode is player turn
+var currentPlayer = "playerTurn";
+//Player & Computer cards in an array
+playerCards = [];
+computerCards = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  currentPlayer = "playerTurn";
+  var deck = shuffleCards(makeDeck());
+  var playerCounter = 0;
+  var computerCounter = 0;
+
+  //Deal 2 Cards each with deck.pop
+  while (playerCounter < 2) {
+    playerCard = deck.pop();
+    playerCards.push(playerCard);
+    playerCounter = playerCounter + 1;
+  }
+  //If player has drawn two cards, it is computer's turn to draw two cards
+  if (playerCards.length == 2) {
+    currentPlayer = "computerTurn";
+    while (computerCounter < 2) {
+      computerCard = deck.pop();
+      computerCards.push(computerCard);
+      computerCounter = computerCounter + 1;
+    }
```

Comment:
if (playerCards.length == 2) is redundant since you want to pop both player and computer card.  You could combine them both .
var index = 0;
while (index< 2) {
    playerCard = deck.pop();
    playerCards.push(playerCard);
    computerCard = deck.pop();
     computerCards.push(computerCard);
    index= index+ 1;
  }